### ISSUE TITLE
`pack install` doesn't run `postinstall`

### ISSUE THREAD

**Issue by joelberkeley** [stefan-hoeck/idris2-pack#248]
If I `pack install spidr`, pack doesn't run `postinstall`. On the other hand, if I run `pack build spidr.ipkg` locally, it does run `postbuild`.

My script reads a file and prints some output, so it's obvious when it runs

**Comment by joelberkeley** [comment]
I've also checked and `idris2 --install spidr.ipkg` run the `postinstall` script

**Comment by joelberkeley** [comment]
since my postinstall mainly prints to stdout, it's possible that what's happening is that `pack install` is running `postinstall` but is supressing the output

**Comment by stefan-hoeck** [comment]
> since my postinstall mainly prints to stdout, it's possible that `pack install` is running `postinstall` but is supressing the output

Can you try running with `pack --log-level debug` and see whether the expected output is printed? Also, as a default, pack runs `idris2 --install-with-src` internally instead of just `idris2 --install`. Could you please verify if with this command the postinstall hook is being run?

**Comment by joelberkeley** [comment]
`pack --log-level debug` does the trick. Haven't tried the others yet

**Comment by joelberkeley** [comment]
I also just verified that `pack install` without `--log-level debug` does run `postinstall` - I used it to create a file. So it's definitely just suppressing output

**Comment by joelberkeley** [comment]
and yes, `idris2 --install-with-src spidr.ipkg` does run `postinstall`

**Comment by stefan-hoeck** [comment]
> I also just verified that `pack install` without `--log-level debug` does run `postinstall` - I used it to create a file. So it's definitely just suppressing output

Yes, this is the expected behavior. We want to suppress the output from `idris2 --install` (but not from `idris2 --build`), because it clutters the application log with lots of messages about what files get copied where. Hence, the output of postinstall-hooks will also be suppressed.

**Comment by stefan-hoeck** [comment]
I'm closing this, at it describes expected behavior. Feel free to reopen if you want to further discuss this issue.

**Comment by joelberkeley** [comment]
Can I ask why this is the intended behaviour? Why is it different from `postbuild`?

**Comment by joelberkeley** [comment]
btw I don't seem to have the option to reopen the issue

**Comment by joelberkeley** [comment]
OK, thanks for the detailed explanation. I don't want to take up your time over this detail, and am happy to just use `postbuild` for now. However, I'm also interested in working out the right way to do this. To that end, here's what I'm doing:

My `postinstall` simply prints installation instructions appropriate for the specific commit. The instructions just download files and edit shared library links. I have a few reasons for printing the instructions rather than executing them, but eventually I may just execute them, so `postinstall` (or `preinstall`) definitely sounds like the right place for that.

**Comment by joelberkeley** [comment]
I like the last idea of `idris --install` logging at different levels so it's possible to silence some parts but not others. I definitely think the output of `postinstall` and the list of which files were copied where could be at different logging levels. I guess it depends on how library authors use `postinstall` and `postbuild`, which could contain useful stuff, like my instructions, but it could also contain a whole load of noise.

**Comment by buzden** [comment]
As a workaround (if it is appropriate for you), you can print instructions to `stderr` in your `postinstall`. In this case, they won't be suppressed by `pack`.

**Comment by joelberkeley** [comment]
thanks. For now I've used `postbuild`

**Comment by joelberkeley** [comment]
I feel it's worth addressing this somehow in some way but don't expect to need it myself so don't need the issue to stay open

**Comment by joelberkeley** [comment]
For anyone looking at this, I notice there are configurable log levels in the system pack.toml. Might help, haven't tried.
